packages/did-ethr-resolver/src/implementations/ewJsonRcpSigner.ts

```diff
@@ -1,23 +1,32 @@
 import { providers } from 'ethers';
 import { EwSigner } from './ewSigner';
 
-type Web3ProviderType = providers.ExternalProvider | providers.JsonRpcFetchFunc;
+type EIP1193ProviderType = providers.ExternalProvider | providers.JsonRpcFetchFunc;
 
 /**
  * An encapsulation of an ethers JsonRpcSigner.
  * This is to allow consumers more flexibility in the ethers version they can use.
  */
 export class EwJsonRpcSigner extends EwSigner {
   /**
-   * Allows the creation of EwJsonRpcSigner using an ethers Provider.
-   * This is convenient if a suitable ethers provider is available
-   * @param provider an ethers JsonPrcProvider or Web3Provider, with an associated signer
+   * A private constructor as this class uses factory method for instantiation API.
    */
-  constructor(provider: providers.JsonRpcProvider | providers.Web3Provider) {
+  private constructor(provider: providers.JsonRpcProvider | providers.Web3Provider) {
     const signer = provider.getSigner();
     super(signer);
   }
 
+  /**
+   * Allows the creation of EwJsonRpcSigner using an ethers library Provider.
+   * This is convenient if a suitable ethers provider is available.
+   * If instead an EIP1993 provider is available, see {@linkcode fromEIP1193}
+   * @param provider an ethers JsonPrcProvider or Web3Provider, with an associated signer
+   */
+  // eslint-disable-next-line max-len
+  public static fromEthersProvider(provider: providers.JsonRpcProvider | providers.Web3Provider): EwJsonRpcSigner {
+    return new EwJsonRpcSigner(provider);
+  }
```

Comment:
This factory method is a little redundant because it simply delegates to the constructor, but it is maybe less confusing for consumers of the API to know to look for factory methods for creation instead of mixing constructor and factory method

Reply:
What do you think @JGiter @knzeng-e?

Reply:
Let it be. Constructor might be changed later on. Also I don't really understand purpose of `EwJsonRpcSigner` and `IdentityOwner`. Could you please elaborate what prevents from defining `fromPrivateKey`, `fromEthersProvider` and `fromEIP1193Provider` directly on `EwSigner`?
And what is the reason of `JsonRpcProvider | Web3Provider` type?

Reply:
- I think we could put `fromEthersProvider` and `fromEIP1193Provider` directly on `EwSigner`. However, `EwPrivateKeySigner` has additional `public` and `private` key properties. If that is not useful to have then we can also put `fromPrivateKey` on `EwSigner`.
- `IdentityOwner` extends `EwSigner` with a `publicKey` property. That is its purpose from my perspective.
- Ah, you're right, I see that `Web3Provider` extends `JsonRpcProvider`, so we can remove `Web3Provider` from the type: https://github.com/ethers-io/ethers.js/blob/f2a32d0d5b4ea3721d3f3ee14db56e0519cf337f/packages/providers/src.ts/web3-provider.ts#L136